sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Testing/EventHubScope.cs

```diff
@@ -94,15 +108,61 @@ public async ValueTask DisposeAsync()
+            {
+                IPage<ConsumerGroup> consumerGroups = client.ConsumerGroups.ListByEventHub(TestEnvironment.EventHubsResourceGroup,
+                                                                                           TestEnvironment.EventHubsNamespace,
+                                                                                           TestEnvironment.EventHubName);
+
+                return new EventHubScope(TestEnvironment.EventHubName, consumerGroups.Select(c => c.Name).ToList(), wasCreated: false);
+            }
+        }
+
+        /// <summary>
+        ///   Performs the tasks needed to get or create a new Event Hub instance with the requested
         ///   partition count and a dynamically assigned unique name.
         /// </summary>
         ///
         /// <param name="partitionCount">The number of partitions that the Event Hub should be configured with.</param>
         /// <param name="caller">The name of the calling method; this is intended to be populated by the runtime.</param>
         ///
         public static Task<EventHubScope> CreateAsync(int partitionCount,
-                                                      [CallerMemberName] string caller = "") => CreateAsync(partitionCount, Enumerable.Empty<string>(), caller);
+                                                      [CallerMemberName] string caller = "") => CreateIfMissingAsync(partitionCount, Enumerable.Empty<string>(), caller);
+
+        /// <summary>
+        ///   It returns the eventhub that is named using the 'EVENT_HUBS_EVENT_HUB_NAME'
+        ///   environment variable. It creates and returns one otherwise.
+        ///
+        ///   If a hub was created, it would be disposed afterwards.
+        /// </summary>
+        ///
+        /// <param name="partitionCount">The number of partitions that the Event Hub should be configured with.</param>
+        /// <param name="consumerGroups">The set of consumer groups to create and associate with the Event Hub; the default consumer group should not be included, as it is implicitly created.</param>
+        /// <param name="caller">The name of the calling method; this is intended to be populated by the runtime.</param>
+        ///
+        /// <returns>The <see cref="EventHubScope" /> that will be used in a given test run.</returns>
+        ///
+        private static Task<EventHubScope> CreateIfMissingAsync(int partitionCount,
+                                                                IEnumerable<string> consumerGroups,
+                                                                [CallerMemberName] string caller = "")
+        {
+            if (!string.IsNullOrEmpty(TestEnvironment.EventHubName))
+            {
+                return PopulateAsync();
+            }
+
+            return CreateAsync(partitionCount, consumerGroups, caller);
```

Comment:
Just realized some tests are invoking it directly and doing so they would create their own Event Hub even if one had already been set from Environment?

Did the change by overloading ```CreateAsync``` and renaming the existing ```CreateAsync``` into ```CreateEventHubAsync```